app/forms/update_user_mobile_form.rb

```diff
@@ -0,0 +1,41 @@
+class UpdateUserMobileForm
+  include ActiveModel::Model
+  include FormMobileValidator
+
+  attr_accessor :mobile
+  attr_reader :user
+
+  def persisted?
+    true
+  end
+
+  def initialize(user)
+    @user = user
+    self.mobile = @user.mobile
+  end
+
+  def submit(params)
+    process_attributes(params)
+
+    params.delete('mobile') if mobile_changed?
+
+    return unless valid?
+    @user.update(params)
```

Comment:
Since mobile is the only possible param here, only the first line of this method will actually do anything.

So I would simply inline the process_attributes method here.   There is probably more simplification that is possible too.